### ISSUE TITLE
Chunk Thrashing in 1.18

### ISSUE THREAD

**Issue by DarkArc** [PaperSpigot/Paper#7389]
### Expected behavior

Chunks don't "thrash" (load only to immediately unload).

### Observed/Actual behavior

Chunks are loading, being unloaded, then loaded/unloaded quickly.

### Steps/models to reproduce

Run around a server with something to monitor chunk loads.

### Plugin and Datapack List

```
Plugins (21): ChestShop, CommandBook, CraftBook, LogBlock, LogBlockQuestioner*, LuckPerms, LWC, MCMailBox, Minequery*, MobArena, Multiverse-Core, Pitfall, PLGBarrierGen, PLGCityGen, PLGVoidGen, ProtocolLib, spark, Vault, WorldEdit, WorldGuard, Ziggy
```

### Paper version

```
[23:50:44 INFO]: This server is running Paper version git-Paper-168 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT) (Git: dcca6cb)
You are running the latest version
Previous version: git-Paper-142 (MC: 1.18.1)
```

### Other

I think this is the guilty code, I'm not sure I understand what it's trying to do?

```
        // ChunkUnloadEvent: Called before the chunk is unloaded: isChunkLoaded is still true and chunk can still be modified by plugins.
        if (playerchunk_state.isOrAfter(ChunkHolder.FullChunkStatus.BORDER) && !playerchunk_state1.isOrAfter(ChunkHolder.FullChunkStatus.BORDER)) {
            this.getFutureIfPresentUnchecked(ChunkStatus.FULL).thenAccept((either) -> {
                io.papermc.paper.util.TickThread.ensureTickThread("Async full status chunk future completion"); // Paper
                LevelChunk chunk = (LevelChunk)either.left().orElse(null);
                if (chunk != null && chunk.wasLoadCallbackInvoked() && ChunkHolder.this.ticketLevel > 33) { // Paper - only invoke unload if load was called
                    // Paper  start - only schedule once, now the future is no longer completed as RIGHT if unloaded...
                    if (ChunkHolder.this.unloadCallbackScheduled) {
                        return;
                    }
                    ChunkHolder.this.unloadCallbackScheduled = true;
                    // Paper  end - only schedule once, now the future is no longer completed as RIGHT if unloaded...
                    chunkStorage.callbackExecutor.execute(() -> {
                        // Paper  start - only schedule once, now the future is no longer completed as RIGHT if unloaded...
                        ChunkHolder.this.unloadCallbackScheduled = false;
                        if (ChunkHolder.this.ticketLevel <= 33) {
                            return;
                        }
                        // Paper  end - only schedule once, now the future is no longer completed as RIGHT if unloaded...
                        // Minecraft will apply the chunks tick lists to the world once the chunk got loaded, and then store the tick
                        // lists again inside the chunk once the chunk becomes inaccessible and set the chunk's needsSaving flag.
                        // These actions may however happen deferred, so we manually set the needsSaving flag already here.
                        chunk.setUnsaved(true);
                        chunk.unloadCallback();
                    });
                }
            }).exceptionally((throwable) -> {
                // ensure exceptions are printed, by default this is not the case
                MinecraftServer.LOGGER.fatal("Failed to schedule unload callback for chunk " + ChunkHolder.this.pos, throwable);
                return null;
            });
```

From my own checker:

```
[23:41:44 WARN]:        at gg.packetloss.grindstone.admin.ProblemDetectionComponent$ChunkMonitor.registerChunkActivity(ProblemDetectionComponent.java:212)
[23:41:44 WARN]:        at gg.packetloss.grindstone.admin.ProblemDetectionComponent$ChunkMonitor.onServerTick(ProblemDetectionComponent.java:224)
[23:41:44 WARN]:        at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:37)
[23:41:44 WARN]:        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76)
[23:41:44 WARN]:        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[23:41:44 WARN]:        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628)
[23:41:44 WARN]:        at net.minecraft.world.level.chunk.Chunk.loadCallback(Chunk.java:809)
[23:41:44 WARN]:        at net.minecraft.server.level.PlayerChunk.lambda$updateFutures$20(PlayerChunk.java:679)
[23:41:44 WARN]:        at net.minecraft.server.level.PlayerChunkMap$CallbackExecutor.run(PlayerChunkMap.java:195)
[23:41:44 WARN]:        at net.minecraft.server.level.ChunkProviderServer$b.y(ChunkProviderServer.java:1260)
[23:41:44 WARN]:        at net.minecraft.server.level.ChunkProviderServer.d(ChunkProviderServer.java:831)
[23:41:44 WARN]:        at net.minecraft.server.MinecraftServer.bf(MinecraftServer.java:1401)
[23:41:44 WARN]:        at net.minecraft.server.MinecraftServer.y(MinecraftServer.java:1384)
[23:41:44 WARN]:        at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:132)
[23:41:44 WARN]:        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1362)
[23:41:44 WARN]:        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1268)
[23:41:44 WARN]:        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317)
[23:41:44 WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by Malfrador** [comment]
I am unable to reproduce this on 1.20.1. Chunks are unloaded and loaded only once, as expected.
Considering the chunk system has changed since 1.18.1 and the suspected code snippet you provided no longer exists, I am going to close this issue. 

If you are able to reproduce this issue on latest (1.20.1) without any plugins other than a testing plugin, feel free to open a new issue.
